src/envs/behavior.py

```diff
@@ -68,7 +68,14 @@ def __init__(self) -> None:
         # a map between task nums and the snapshot id for saving/loading
         # purposes
         self.task_num_to_igibson_seed: Dict[int, int] = {}
-
+        # Everytime we load a new scene in behavior we also need to set
+        # the vaild options again, because there might be new type combos
+        self.set_options()
+
+    def set_options(self) -> None:
+        """Sets the underlying options for this particular task using the
+        current type list."""
+        self._type_name_to_type = {}
```

Comment:
this line seems unrelated to setting the options. if the idea is to clear this dict every time the scene is reset, let's add this line into a reset method?